### PR TITLE
feat: add other policies in CEL expressions - Part 1

### PR DESCRIPTION
**Pull request by Chandan-DK**
## Related Issue(s)
Partially addresses #891 

## Description
This PR includes the conversion of policies in the other folder to Kyverno CEL policies. Conversion of the policies will be done in multiple PRs

**Policies converted in this PR:**

- [x] allowed-annotations
- [x] allowed-pod-priorities
- [x] block-ephemeral-containers
- [x] check-env-vars
- [x] check-node-for-cve-2022-0185
- [x] check-serviceaccount-secrets
- [x] deny-secret-service-account-token-type
- [x] disallow-all-secrets
- [x] disallow-localhost-services
- [x] disallow-secrets-from-env-vars
- [x] docker-socket-requires-label

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [x] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

File: other-cel/allowed-pod-priorities/allowed-pod-priorities.yaml
Comment (MariamFahmy98) [inline other-cel/allowed-pod-priorities/allowed-pod-priorities.yaml:1]: Can we use the autogen rules in this policy?

File: other-cel/check-node-for-cve-2022-0185/check-node-for-cve-2022-0185.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  - name: kernel-validate
+    match:
+      any:
+      - resources:
+          kinds:
+            - Node
+    validate:
+      cel:
+        expressions:
+          - expression: "!(object.status.nodeInfo.kernelVersion in ['5.10.84-1', '5.15.5-2'])"
```
Comment (MariamFahmy98) [inline other-cel/check-node-for-cve-2022-0185/check-node-for-cve-2022-0185.yaml:32]: Have you tried this policy? I am not sure if we could access the `status` field.

**Review by MariamFahmy98 — COMMENTED** [review]
There is a missing kyverno test in the `allowed-pod-priorities` and `block-ephemeral-containers`.

File: other-cel/check-node-for-cve-2022-0185/check-node-for-cve-2022-0185.yaml (hunk 1 above)
Comment (Chandan-DK) [reply other-cel/check-node-for-cve-2022-0185/check-node-for-cve-2022-0185.yaml:32]: I'm running a single node kind cluster. This is the node's configuration:
```
$ k get nodes -o jsonpath='{.items[*].status.nodeInfo}' | jq .
{
  "architecture": "amd64",
  "bootID": "684deeaf-72c0-463c-9a17-c24cc3d15391",
  "containerRuntimeVersion": "containerd://1.7.13",
  "kernelVersion": "6.5.0-1015-aws",
  "kubeProxyVersion": "v1.29.2",
  "kubeletVersion": "v1.29.2",
  "machineID": "1098f0df08574920a984b8f198a89076",
  "operatingSystem": "linux",
  "osImage": "Debian GNU/Linux 12 (bookworm)",
  "systemUUID": "35178929-44c5-4ff4-bf17-8e84eb1f3f48"
}
```

After applying the policy, this is the generated VAP:
```
$ k get validatingadmissionpolicy -o yaml
apiVersion: v1
items:
- apiVersion: admissionregistration.k8s.io/v1beta1
  kind: ValidatingAdmissionPolicy
  metadata:
    creationTimestamp: "2024-03-26T09:55:13Z"
    generation: 1
    labels:
      app.kubernetes.io/managed-by: kyverno
    name: check-kernel
    ownerReferences:
    - apiVersion: kyverno.io/v1
      kind: ClusterPolicy
      name: check-kernel
      uid: ba1a0d27-5466-4686-b0da-a9efa6f4209b
    resourceVersion: "2251"
    uid: d5f8d499-8b1e-46ca-b069-171136d9307a
  spec:
    failurePolicy: Fail
    matchConstraints:
      matchPolicy: Equivalent
      namespaceSelector: {}
      objectSelector: {}
      resourceRules:
      - apiGroups:
        - ""
        apiVersions:
        - v1
        operations:
        - CREATE
        - UPDATE
        resources:
        - nodes
        scope: '*'
    validations:
    - expression: '!(object.status.nodeInfo.kernelVersion in [''5.10.84-1'', ''5.15.5-2''])'
      message: Kernel is vulnerable to CVE-2022-0185.
  status:
    observedGeneration: 1
    typeChecking: {}
kind: List
metadata:
  resourceVersion: ""
```
This is the generated report we get:
```
$ k get cpolr -o yaml
apiVersion: v1
items:
- apiVersion: wgpolicyk8s.io/v1alpha2
  kind: ClusterPolicyReport
  metadata:
    creationTimestamp: "2024-03-26T09:55:24Z"
    generation: 1
    labels:
      app.kubernetes.io/managed-by: kyverno
    name: fe7942a7-da9b-4390-aaa6-a869a4accb13
    ownerReferences:
    - apiVersion: v1
      kind: Node
      name: vap-cluster-control-plane
      uid: fe7942a7-da9b-4390-aaa6-a869a4accb13
    resourceVersion: "2292"
    uid: fd5b66b9-5aa3-4b69-abc4-ce7f4d1744ac
  results:
  - category: Other in CEL
    message: Validation rule 'kernel-validate' passed.
    policy: check-kernel
    result: pass
    rule: kernel-validate
    scored: true
    severity: high
    source: kyverno
    timestamp:
      nanos: 0
      seconds: 1711446914
  scope:
    apiVersion: v1
    kind: Node
    name: vap-cluster-control-plane
    uid: fe7942a7-da9b-4390-aaa6-a869a4accb13
  summary:
    error: 0
    fail: 0
    pass: 1
    skip: 0
    warn: 0
kind: List
metadata:
  resourceVersion: ""
```
We are able to access the status field here.

File: other/deny-secret-service-account-token-type/artifacthub-pkg.yaml [hunk 2]
```diff
@@ -26,5 +26,5 @@ readme: |
 annotations:
   kyverno/category: "Security"
   kyverno/kubernetesVersion: "1.27"
-  kyverno/subject: "Secret,ServiceAccount"
+  kyverno/subject: "Secret"
```
[the remark this answers is not in the document]
Comment (Chandan-DK) [reply other/deny-secret-service-account-token-type/artifacthub-pkg.yaml]: I did it to make it consistent with the policy's corresponding [annotation](https://github.com/kyverno/policies/blob/main/other/deny-secret-service-account-token-type/deny-secret-service-account-token-type.yaml#L11) which only has `Secret`. I should have added `ServiceAccount` to the policy's annotation instead. I have made the changes.

**Comment by Chandan-DK** [comment]
> There is a missing kyverno test in the `allowed-pod-priorities` and `block-ephemeral-containers`.

We can't have kyverno tests for `allowed-pod-priorities` at the moment because support for parameter resources in CLI tests has to be added ([Issue](https://github.com/kyverno/kyverno/issues/9431))

File: other-cel/allowed-pod-priorities/allowed-pod-priorities.yaml
Comment (Chandan-DK) [reply other-cel/allowed-pod-priorities/allowed-pod-priorities.yaml:1]: You are right. I have made the changes.

Comment (Chandan-DK) [reply other-cel/allowed-pod-priorities/allowed-pod-priorities.yaml:1]: I am making changes in the corresponding Kyverno policy to use autogen. I am looking at what changes have to be made in the kyverno tests for it. Will resolve this conversation once it's done.

File: other-cel/check-node-for-cve-2022-0185/check-node-for-cve-2022-0185.yaml (hunk 1 above)
Comment (MariamFahmy98) [reply other-cel/check-node-for-cve-2022-0185/check-node-for-cve-2022-0185.yaml:32]: Great!

**Comment by MariamFahmy98** [comment]
It seems that there are some flake tests. Could you please check?

**Comment by Chandan-DK** [comment]
Sure. Will take a look soon

**Review by MariamFahmy98 — APPROVED** [review]
Looks good to me. Thank you!

**Comment by JimBugwadia** [comment]
@Chandan-DK - can you please help resolve the conflicts?

**Comment by Chandan-DK** [comment]
> @Chandan-DK - can you please help resolve the conflicts?

Sure 👍

**Comment by Chandan-DK** [comment]
The conflicts have been resolved

**Pull request merged**